presto-tests/src/test/java/com/facebook/presto/tests/TestQueryTaskLimit.java

```diff
@@ -135,4 +149,32 @@ private void waitForQueryToBeKilled(DistributedQueryRunner queryRunner)
             MILLISECONDS.sleep(10);
         }
     }
+
+    private void waitForQueryToBeCancelled(DistributedQueryRunner queryRunner, QueryId queryId)
```

Comment:
Use `TestQueryRunnerUtil.waitForQuerState`?

Reply:
TestQueryRunnerUtil.waitForQueryState() can only check the query's state has changed to FAILED. Is it enough to make sure that the query is actually been cancelled rather than timed out?

Reply:
I think that's fine.

Reply:
OK, I have fixed it.